File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestMonotonicSubarray = function(nums) {
    let countr = 1;
    let left = 0;
    let countl = 1;

    for(let right=1;right<nums.length;right++){
        if(nums[right] > nums[right-1]){
            countl = Math.max(countl, right-left+1);
            if(nums[right+1] <= nums[right])
            left = right;
        }

        else if(nums[right] < nums[right-1]){
            countr = Math.max(countr, right-left+1);
            if(nums[right+1] >= nums[right])
            left = right;
        }

        else{
            left = right;
        }
    }
    return Math.max(countl, countr);
};

File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.java
class Solution {
    public int longestMonotonicSubarray(int[] nums) {
        int increase = 1;
        int decrease = 1;
        int result = 1;

        for(int i=1;i<nums.length;i++){
            if(nums[i] > nums[i-1])
            {
                increase++;
                decrease = 1;
            }

            else if(nums[i] < nums[i-1]){
                decrease++;
                increase = 1;
            }

            else{
                decrease = 1;
                increase = 1;
            }

            result = Math.max(result, Math.max(increase, decrease));
        }
        return result;
    }
}